query/src/sql/planner/plans/logical_join.rs

```diff
@@ -23,15 +23,27 @@ use crate::sql::plans::PhysicalPlan;
 use crate::sql::plans::PlanType;
 use crate::sql::plans::Scalar;
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum JoinType {
+    InnerJoin,
+    LeftJoin,
+    RightJoin,
+    FullJoin,
+    SemiJoin,
+    AntiJoin,
+    CrossJoin,
+}
+
 #[derive(Clone, Debug)]
-pub struct LogicalInnerJoin {
+pub struct LogicalJoin {
     pub left_conditions: Vec<Scalar>,
     pub right_conditions: Vec<Scalar>,
+    pub join_type: JoinType,
```

Comment:
It's better to introduce another struct `CrossJoin`.

The predicate of inner join and outer join are different. 

We can split conditions into `left_conditions` and `right_conditions` for inner join, but for outer join, we have to keep the `other_conditions` too.

And for `CrossJoin`, we will lift the predicates into a `Filter` for push down.